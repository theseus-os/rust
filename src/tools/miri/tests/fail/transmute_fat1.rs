#[cfg(target_pointer_width = "64")]
const N: usize = 16;

#[cfg(target_pointer_width = "32")]
const N: usize = 8;

fn main() {
    let bad = unsafe {
        std::mem::transmute::<&[u8], [u8; N]>(&[1u8])
        //~^ ERROR: constructing invalid value: encountered a pointer
    };
    let _val = bad[0] + bad[bad.len() - 1];
}
