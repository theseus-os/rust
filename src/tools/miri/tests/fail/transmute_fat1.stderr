error: Undefined Behavior: constructing invalid value: encountered a pointer, but expected plain (non-pointer) bytes
  --> $DIR/transmute_fat1.rs:LL:CC
   |
LL |         std::mem::transmute::<&[u8], [u8; N]>(&[1u8])
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ constructing invalid value: encountered a pointer, but expected plain (non-pointer) bytes
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: backtrace:
   = note: inside `main` at $DIR/transmute_fat1.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

